test/CachedQuery.test.ts

```diff
@@ -1,311 +1,235 @@
+        const query = queryShallow(fetch, refetch);
+        const result = await query.run(1).run();
+        fetch.assertExhausted();
+        expect(result).toEqual(right(2));
       });
 
       it('viene chiamata una volta una fetch che fallisce sempre e run ritorna il fallimento', async () => {
-        const fetch = new StatefulFetch({
-          order: 'alwaysFailure',
-          resultType: 'alwaysTheSame'
-        }).fetch;
-        const { fetchFunction, result } = await runQueryShallow(
-          fetch,
-          1,
-          refetch
-        );
-        expect(fetchFunction.mock.calls.length).toBe(1);
-        expect(result.isLeft()).toBe(true);
+        const fetch = makeTestFetch([() => fromLeft('nope')]);
+        const query = queryShallow(fetch, refetch);
+        const result = await query.run().run();
+        fetch.assertExhausted();
+        expect(result).toEqual(left('nope'));
       });
 
       it('dopo avere eseguito run con successo una volta, viene chiamata di nuovo run e la fetch esegue nuovamente con successo ritornando il nuovo risultato', async () => {
-        const fetch = new StatefulFetch({
-          order: 'alwaysSuccess',
-          resultType: 'alwaysDifferent'
-        }).fetch;
-        const { fetchFunction, result, query } = await runQueryShallow(
-          fetch,
-          1,
-          refetch
-        );
-        expect(fetchFunction.mock.calls.length).toBe(1);
-        expect(result.isRight()).toBe(true);
-        const result2 = await query.run(1).run();
-        expect(fetchFunction.mock.calls.length).toBe(2);
-        expect(result2.isRight()).toBe(true);
-        expect(result2.getOrElse('') !== result.getOrElse('')).toBe(true);
+        const fetch = makeTestFetch([
+          (n: number) => taskEither.of(n * 2),
+          (n: number) => taskEither.of(n / 2)
+        ]);
+        const query = queryShallow(fetch, refetch);
+        const result = await query.run(2).run();
+        expect(result).toEqual(right(4));
+        const result2 = await query.run(2).run();
+        expect(result2).toEqual(right(1));
+        fetch.assertExhausted();
       });
 
       it('dopo avere eseguito run con successo una volta, viene chiamata di nuovo run e la fetch esegue nuovamente fallendo e ritornando il fallimento', async () => {
-        const fetch = new StatefulFetch({
-          order: 'successFirst',
-          resultType: 'alwaysDifferent'
-        }).fetch;
-        const { fetchFunction, result, query } = await runQueryShallow(
-          fetch,
-          1,
-          refetch
-        );
-        expect(fetchFunction.mock.calls.length).toBe(1);
-        expect(result.isRight()).toBe(true);
+        const fetch = makeTestFetch([
+          (n: number) => taskEither.of(n * 2),
+          () => fromLeft('nope')
+        ]);
+        const query = queryShallow(fetch, refetch);
+        const result = await query.run(1).run();
+        expect(result).toEqual(right(2));
```

Comment:
this test on `result` looks redundant now that I look at it, what do you think?

Reply:
pushed a commit where all intermediate result checks are removed